File: app/src/main/java/com/example/myapplication/domain/usecase/ObserveUserServiceConnectionUseCase.java
package com.example.myapplication.domain.usecase;

import com.example.myapplication.domain.repository.ConnectionStateListener;
import com.example.myapplication.domain.repository.UserRepository;

public class ObserveUserServiceConnectionUseCase {

    private final UserRepository userRepository;

    public ObserveUserServiceConnectionUseCase(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public void register(ConnectionStateListener listener) {
        userRepository.addConnectionStateListener(listener);
    }

    public void unregister(ConnectionStateListener listener) {
        userRepository.removeConnectionStateListener(listener);
    }
}

File: app/src/main/java/com/example/myapplication/domain/usecase/ObserveUserServiceConnectionUseCase.kt
package com.example.myapplication.domain.usecase

import com.example.myapplication.domain.repository.ConnectionStateListener
import com.example.myapplication.domain.repository.UserRepository

class ObserveUserServiceConnectionUseCase(
    private val userRepository: UserRepository,
) {
    fun register(listener: ConnectionStateListener) {
        userRepository.addConnectionStateListener(listener)
    }

    fun unregister(listener: ConnectionStateListener) {
        userRepository.removeConnectionStateListener(listener)
    }
}
